main.py

```diff
@@ -1,12 +1,393 @@
 import requests
+import pyfiglet
+import re
+# import itertools, threading, time, sys
+from customers import Customers
+from videos import Videos
+from rentals import Rentals
 
-URL = "http://127.0.0.1:5000"
-BACKUP_URL = "https://retro-video-store-api.herokuapp.com"
+URL = "https://whits-video-store.herokuapp.com/"
 
+def print_stars():
+    print("\n*******************************\n")
+
+def list_options():
+    video_options = {
+        "1": "Get detailed information for all videos",
+        "2": "Look up video",
+        "3": "Look up videos by genre",
+        "4": "Add a video to the inventory",
+        "5": "Edit video information",
+        "6": "Delete a video from the inventory"
+    }
+
+    customer_options = {
+        "7": "Get detailed information for all customers",
+        "8": "Look up customer",
+        "9": "Add a customer to the database",
+        "10": "Edit customer information",
+        "11": "Delete a customer from the database",
+        "12": "Look up a customer's checked-out videos"
+    }
+
+    rental_options = {
+        "13": "Check out a video",
+        "14": "Check in a video",
+        "15": "List current rentals",
+        "16": "List overdue rentals",
+    }
+
+
+    print_stars()
+    menu_heading = pyfiglet.figlet_format("MAIN MENU", font = "contessa")
+    print(menu_heading)
+
+    video_heading = pyfiglet.figlet_format("VIDEO ACTIONS", font = "digital")
+    print(video_heading)
+    for choice_num in video_options:
+        print(f"Option {choice_num}. {video_options[choice_num]}")
+
+    customer_heading = pyfiglet.figlet_format("CUSTOMER ACTIONS", font = "digital")
+    print(f"\n{customer_heading}")
+    for choice_num in customer_options:
+        print(f"Option {choice_num}. {customer_options[choice_num]}")
+
+    rental_heading = pyfiglet.figlet_format("RENTAL ACTIONS", font = "digital")
+    print(f"\n{rental_heading}")
+    for choice_num in rental_options:
+        print(f"Option {choice_num}. {rental_options[choice_num]}")
+
+    print_stars()
+
+    all_options = {**video_options, **customer_options, **rental_options}
+    return all_options
+
+
+def make_choice(options):
+    valid_choices = options.keys()
+    other_valid_choices = ["o", "x"]
+    choice = None
+
+    while choice not in valid_choices and choice not in other_valid_choices:
+        print("What would you like to do? Type 'o' to see all options. Type 'x' to exit store.")
+        choice = input("Make your selection using the option number: ")
+
+    return choice
+
+
+def run_cli(play=True):
+
+    customers = Customers()
+    videos = Videos()
+    rentals = Rentals()
```

Comment:
Notice that these 3 classes do the same thing.  Instead of 3 classes, you could make 1 class who's role is to interact with the API.  Then the `main.py` can handle interactions with the user with print and input.